File: tangibles/spheros/spheros/src/communication/AbsTcpCommunication.java
package communication;

import java.io.BufferedInputStream;
import java.io.BufferedOutputStream;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.net.Socket;
import java.util.logging.Level;
import java.util.logging.Logger;

public class AbsTcpCommunication extends AbsCommunication
{
	protected Socket _sock;
	protected BufferedInputStream _in;
	protected BufferedOutputStream _out;

	protected OutputStreamWriter _outString;

	public AbsTcpCommunication( Socket sock ) throws IOException
	{
		this._sock = sock;
		_in = new BufferedInputStream( sock.getInputStream() );
		_out = new BufferedOutputStream( sock.getOutputStream() );

		_outString = new OutputStreamWriter( sock.getOutputStream() );
	}

	public Socket getSocket()
	{
		return _sock;
	}
	
	public void sendBytes( byte[] bytes ) throws IOException
	{
		_out.write( bytes, 0, bytes.length );
		_out.flush();
	}

	public void sendString( String msg ) throws IOException
	{
		Logger.getLogger( this.getClass().getCanonicalName() ).log( Level.INFO, "Sending << " + msg + ">>" );

		_outString.write( msg.toCharArray() );
		_outString.flush();

		Logger.getLogger( this.getClass().getCanonicalName() ).log( Level.INFO, "<< Message sent >>" );
	}
}

File: tangibles/SiftDriver/SiftDriver/Communication/AbsTcpCommunication.cs
using System;
using System.Net.Sockets;
using System.IO;

namespace SiftDriver.Communication
{
	public class AbsTcpCommunication :AbsCommunication
	{
    private TcpClient _sock;
    private NetworkStream _socketStream;
    private TextReader _in;
    private TextWriter _out;
    public TextWriter Output{ get { return _out; } }
    public TextReader Input{ get {return _in;} }


		public AbsTcpCommunication (TcpClient sock)
		{
      _sock = sock;
			_socketStream = _sock.GetStream();
      _in = new StreamReader(_socketStream);
      _out = new StreamWriter(_socketStream);
		}

    public void sendBytes(byte[] bytes){
      _socketStream.Write(bytes, 0, bytes.Length);
      _socketStream.Flush();
    }
	}
}

